fri/src/prover/mod.rs

```diff
@@ -222,12 +213,12 @@ where
         });
     }
 
-    /// Creates a FriRemainder from a vector of `evaluations` representing the remainder and
-    /// commits to the latter by hashing sequentially `evaluations`.
-    fn set_remainder(&mut self, channel: &mut C, evaluations: &mut [E]) {
-        let commitment = <H as ElementHasher>::hash_elements(evaluations);
-        channel.commit_fri_layer(commitment);
-        self.remainder = FriRemainder(evaluations.to_vec());
+    /// Creates remainder polynomial in coefficient form from a vector of `evaluations` over a domain.
+    fn set_remainder(&mut self, evaluations: &mut [E]) {
+        let inv_twiddles = fft::get_inv_twiddles(evaluations.len());
+        fft::interpolate_poly_with_offset(evaluations, &inv_twiddles, E::BaseField::GENERATOR);
+        let remainder = evaluations[..evaluations.len() / self.options.blowup_factor()].to_vec();
+        self.remainder = FriRemainder(remainder);
     }
```

Comment:
A few comments/questions here:
- Should we get the domain offset from options here rather than using `E::BaseField::GENERATOR`?
- One of the things that we did in `commit_fri_layer()` was reseeding of the random coin. I think we might still need to do this, but now we should reseed it with the entire remainder polynomial - right? If so, it might be more efficient to commit to the remainder polynomial and then use this commitment to reseed the public coin with this commitment.
- minor nit: I'd probably rename `remainder` variable into `remainder_poly`.